Added support for ChibiOS clients

stertingen opened the pull request.

stertingen committed:
Fix warning when using std_msgs/Empty (#482)

stertingen committed:
ros_lib: Removed code duplication in node_handle.h (#483)

* ros_lib: Removed code duplication in node_handle.h

Also, do not generate multiple template instances of
NodeHandle::subscribe()

* Compilefix: call subscribe() with reference, not pointer

stertingen committed:
Added support for ChibiOS clients

stertingen committed:
Do not fail when trying to update ros_lib which already exists (#494)

Co-authored-by: Mike Purvis <mpurvis@clearpath.ai>

mikepurvis commented on rosserial_chibios/src/rosserial_chibios/make_libraries.py:
```diff
@@ -0,0 +1,94 @@
+
+# generate messages
+rosserial_generate(rospack, path+"/ros_lib", ROS_TO_EMBEDDED_TYPES)
+
```
I know this is just a c&p of similar other files in this repo, but please lose the trailing line and add spaces around the `+` operators above (or better yet, switch them to be `os.path.join`).

stertingen replied:
So be it.

`os.path.join` strips trailing slashes from the path; which breaks further generation of `ros_lib`.

There'll be a PR to migrate everything to `os.path.join` in a minute.

mikepurvis replied:
Ha, appreciated.

stertingen committed:
Use os.path.join for path concatenation

mikepurvis commented:
Retargeting this to `noetic-devel` and marking for backport consideration.

stertingen committed:
Bump required CMake version

stertingen committed:
Merge branch 'noetic-devel' into rosserial_chibios

stertingen committed:
Prefer OSAL and ChibiOS time conversion macros if defined

OSAL_I2MS was added to OSAL recently, TIME_I2MS is specifig to
ChibiOS/RT/NIL, so keep the fallback solution

mikepurvis commented:
Looks good, thanks!
